boa/src/syntax/lexer/comment.rs

```diff
@@ -0,0 +1,66 @@
+use super::{Cursor, Error, Tokenizer};
+use crate::syntax::ast::{Position, Span};
+use crate::syntax::lexer::{Token, TokenKind};
+use std::io::Read;
+
+pub(super) struct SingleLineComment;
+
+/// Lexes a single line comment.
+///
+/// Assumes that the initial '//' is already consumed.
+///
+/// More information:
+///  - [ECMAScript reference][spec]
+///  - [MDN documentation][mdn]
+///
+/// [spec]: https://tc39.es/ecma262/#sec-comments
+/// [mdn]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar
```

Comment:
I would put the comment in the structure instead of in the implementation, so that it shows nicely when generating the documentation.

Reply:
Done